app/src/main/java/org/systers/mentorship/view/fragments/ChangePasswordFragment.kt

```diff
@@ -54,7 +56,31 @@ class ChangePasswordFragment : DialogFragment() {
         builder.setNegativeButton(getString(R.string.cancel)) { dialog, _ ->
             dialog.cancel()
         }
-        return builder.create()
+
+        val passwordDialog = builder.create()
+
+        passwordDialog.setOnShowListener {
+            passwordDialog?.getButton(AlertDialog.BUTTON_POSITIVE)?.isEnabled = false
+
+            changePasswordView.tilConfirmPassword?.editText?.addTextChangedListener(object : TextWatcher {
+                override fun afterTextChanged(p0: Editable?) {
+                    if (p0!!.isEmpty())
+                        passwordDialog?.getButton(AlertDialog.BUTTON_POSITIVE)?.isEnabled = false
```

Comment:
we are using MVVM pattern here so this is not a good practice. Please try to use binding adapter.

Reply:
@annabauza For that, I believe we need to make the very layout for this fragment declarative for data binding. And then we can enable data binding and it's features for this whole fragment. Let me know your thoughts?

Reply:
I have opened #615. We can enable data binding for the Fragment first, then we can do this change.